lib/js/components/Form/Checkbox/Checkbox.vue

```diff
@@ -1,60 +1,256 @@
+	}
+
+	&__root {
+		background: none;
+		border: none;
+		line-height: 0;
+		margin: 0;
+		padding: 0;
+		position: relative;
+
+		#{$root}.-ds-x-small & {
+			padding: $space-3xs 0;
+		}
+
+		#{$root}.-ds-small & {
+			padding: $space-3xs 0;
+		}
+
+		#{$root}.-ds-medium & {
+			padding: $space-2xs 0;
+		}
+
+		&:focus {
+			outline: none;
+		}
+
+		&::before {
+			aspect-ratio: 1;
+			border-radius: $radius-xl;
+			content: '';
+			display: block;
+			height: 100%;
+			position: absolute;
+			top: 0;
+			transform: scale(0);
+			// see https://cubic-bezier.com/#.23,1,.32,1
+			transition: all $default-transition-time cubic-bezier(0.23, 1, 0.32, 1) 0ms;
+			width: auto;
+		}
+
+		#{$root}:hover:not(.-ds-disabled) &::before,
+		&:focus::before {
+			transform: scale(1);
+		}
+
+		#{$root}.-ds-x-small &::before {
+			left: -6px;
+		}
+
+		#{$root}.-ds-small &::before {
+			left: -6px;
+		}
+
+		#{$root}.-ds-medium &::before {
+			left: -8px;
+		}
+
+		#{$root}:hover &::before {
+			background-color: $color-neutral-background-ghost-hovered;
+		}
+
+		&:focus::before {
+			background-color: $color-neutral-background-ghost-focused;
+		}
+	}
+
+	&__indicator {
+		color: $color-neutral-icon;
+		position: relative;
+
+		#{$root}.-ds-error & {
+			color: $color-danger-icon;
+		}
+
+		#{$root}.-ds-disabled & {
+			color: $color-neutral-icon-disabled;
+		}
+
+		#{$root}.-ds-elevation &::before {
+			background: $color-default-background;
+			bottom: 2px;
+			content: '';
+			left: 2px;
+			position: absolute;
+			right: 2px;
+			top: 2px;
+		}
+
+		&[data-state='checked'] {
+			& #{$root}__unchecked,
+			& #{$root}__indeterminate {
+				display: none;
+			}
+		}
+
+		&[data-state='unchecked'] {
+			& #{$root}__checked,
+			& #{$root}__indeterminate {
+				display: none;
+			}
+		}
+
+		&[data-state='indeterminate'] {
+			& #{$root}__checked,
+			& #{$root}__unchecked {
+				display: none;
+			}
+		}
+	}
+
+	&__label {
+		color: $color-neutral-text-strong;
+
+		#{$root}.-ds-x-small & {
+			@include formLabel-s-default-regular;
+		}
+
+		#{$root}.-ds-small & {
+			@include formLabel-m-default-regular;
+		}
+
+		#{$root}.-ds-medium & {
+			@include formLabel-l-default-regular;
+		}
+
+		#{$root}.-ds-disabled & {
+			color: $color-neutral-text-strong-disabled;
+		}
+	}
+}
+</style>
+
+<script setup lang="ts">
+import { CheckboxRoot, CheckboxIndicator } from 'reka-ui';
+import {
+	CHECKBOX_SIZES,
+	CHECKBOX_STATES,
+	CHECKBOX_ELEVATIONS,
+	CheckboxSize,
+	CheckboxState,
+	CheckboxValue,
+	CheckboxElevation,
+} from './Checkbox.consts';
+import Icon from '../../Icons/Icon/Icon.vue';
+import { ICON_SIZES, ICONS } from '../../Icons/Icon';
+import { computed, inject } from 'vue';
+import { CHECKBOX_GROUP_INJECTION_KEY } from '../CheckboxGroupField/CheckboxGroupField.consts';
+
+const props = defineProps<{
+	size?: CheckboxSize;
+	state?: CheckboxState;
+	elevation?: CheckboxElevation;
+	/**
+	 * String value set when the checkbox is checked
+	 */
+	value: string;
```

Comment:
did you consider setting a default value to `'on'` similar to Reka and native HTML?

Reply:
I marked it as required because we will normally use it inside a checkbox group, where this prop always needs to be provided.

Reply:
I'm not convinced... It's ok that we force this value by `CheckboxGroup` but it's the responsibility of that component to provide the correct value. Because we optimized for that single use case, we have to put a random value in other places.